atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -95,8 +105,18 @@ public void registerMetric(Class clazz, String metricName, Gauge gauge, Map<Stri
         taggedMetricRegistry.gauge(metricToAdd, gauge);
     }
 
-    private Map<String, String> getTableNameTagFor(TableReference tableRef) {
-        return ImmutableMap.of("tableName", LoggingArgs.safeTableOrPlaceholder(tableRef).getQualifiedName());
+    @VisibleForTesting
+    Map<String, String> getTableNameTagFor(TableReference tableRef) {
+        String tableName = tableRef.getTablename();
+        if (!isSafeToLog.apply(tableRef)) {
+            tableName = "unsafeTableRef_" + obfuscate(tableRef);
+        }
+
+        return ImmutableMap.of("tableName", tableName);
+    }
+
+    private String obfuscate(TableReference tableRef) {
+        return Sha256Hash.computeHash(tableRef.getTablename().getBytes()).serializeToHexString().substring(0, 8);
```

Comment:
Was thinking about collisions here, https://stackoverflow.com/questions/19962424/probability-of-collision-with-truncated-sha-256-hash, theres about ~1% probability of collision in 10k tables, but we are fine I think (except for the large internal product that is creating and truncating tables). We can take a 16 character substring?